### PR TITLE
ddl: fix cleanup subtasks after cancel add index with dist-reorg

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/41371

Problem Summary:

### What is changed and how it works?
Do `cleanupBackfillJobs` after canceling the operation.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: sessionctx/variable/featuretag/distributereorg/default.go [hunk 1]
```diff
@@ -17,4 +17,4 @@
 package distributereorg
 
 // TiDBEnableDistributeReorg is a feature tag
-const TiDBEnableDistributeReorg bool = false
+const TiDBEnableDistributeReorg bool = true
```
Comment (Defined2014) [inline sessionctx/variable/featuretag/distributereorg/default.go:20]: We will default enable `DistributeReorg`?

**Comment by hawkingrei** [comment]
@zimulala  I found it in the CI.
```
panic: runtime error: invalid memory address or nil pointer dereference

[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x448b259]


goroutine 183764 [running]:

github.com/pingcap/tidb/ddl.(*reorgCtx).isReorgCanceled(...)

	ddl/reorg.go:113

github.com/pingcap/tidb/ddl.(*reorgCtx).IsReorgCanceled(...)

	ddl/ddl_test.go:62

github.com/pingcap/tidb/ddl_test.TestUsingReorgCtx.func2()

	ddl/ddl_worker_test.go:334 +0xd9

github.com/pingcap/tidb/util.(*WaitGroupWrapper).Run.func1()

	util/wait_group_wrapper.go:154 +0x6b

created by github.com/pingcap/tidb/util.(*WaitGroupWrapper).Run

	util/wait_group_wrapper.go:152 +0x8d

--

Coverage runner: Not collecting coverage for failed test.


```

File: sessionctx/variable/featuretag/distributereorg/default.go (hunk 1 above)
Comment (zimulala) [reply sessionctx/variable/featuretag/distributereorg/default.go:20]: No. It's a mistake.

**Review by Defined2014 — APPROVED** [review]

**Comment by zimulala** [comment]
> @zimulala I found it in the CI.
> 
> ```
> panic: runtime error: invalid memory address or nil pointer dereference
> 
> [signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x448b259]
> 
> 
> goroutine 183764 [running]:
> 
> github.com/pingcap/tidb/ddl.(*reorgCtx).isReorgCanceled(...)
> 
> 	ddl/reorg.go:113
> 
> github.com/pingcap/tidb/ddl.(*reorgCtx).IsReorgCanceled(...)
> 
> 	ddl/ddl_test.go:62
> 
> github.com/pingcap/tidb/ddl_test.TestUsingReorgCtx.func2()
> 
> 	ddl/ddl_worker_test.go:334 +0xd9
> 
> github.com/pingcap/tidb/util.(*WaitGroupWrapper).Run.func1()
> 
> 	util/wait_group_wrapper.go:154 +0x6b
> 
> created by github.com/pingcap/tidb/util.(*WaitGroupWrapper).Run
> 
> 	util/wait_group_wrapper.go:152 +0x8d
> 
> --
> 
> Coverage runner: Not collecting coverage for failed test.
> ```

This may be an unstable test or a bug, but it doesn't because of this PR. I will handle it. Could you add an issue?

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**